common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -409,19 +472,34 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
-                    break;
+                    return DeadlineRunResult.TIMED_OUT;
```

Comment:
This is tricky as `TIMED_OUT` means that the deadline expired, based on which I take decisions in `SingleThreadEventLoop`. If I return `RAN_AT_LEAST_ONCE` here then the caller has to check whether the deadline expired or not. May be I can rename this to `DEADLINE_EXPIRED`?